pkg/engine/virtualmachinescalesets.go

```diff
@@ -14,6 +14,35 @@ import (
 	"github.com/Azure/go-autorest/autorest/to"
 )
 
+func createDiskEncryptionExtensionProperties(isLinux bool) *compute.VirtualMachineScaleSetExtensionProperties {
+	var extensionType, version, volumeType string
+  if isLinux {
+		extensionType = "AzureDiskEncryptionForLinux"
```

Comment:
happened to review this PR, `AzureDiskEncryptionForLinux` is actually not production ready for Linux VM, instead, we are already supporting SSE+CMK to encrypt os and data disk encryption: specify `diskEncryptionSetID` in aks-engine cluster definition file, more details: https://docs.microsoft.com/en-us/azure/aks/azure-disk-customer-managed-keys